js/main.js

```diff
@@ -1,22 +1,109 @@
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg'
+];
 
-// https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/random ссылка на источник!
+const locationLat = {1: 35.65000, 2: 35.70000, 3: 5};
 
-function getRandomFloating (min, max, maxDigits = 0) {
-  if (min > max || min < 0 || max <= 0) {
-    return ('Неверный диапазон значений');
+const locationLng = {1: 139.70000, 2: 139.80000, 3: 5};
+
+const COUNT_ITEMS_OF_OFFER = 10;
+
+/* Старые функции убрал, взял новые у кекса, хотя не уверен в правильности, те немного поправил после твоих заметок
+и они как то локаничней смотрелись!*/
+
+const getRandomPositiveInteger = (a, b) => {
+  const lower = Math.ceil(Math.min(Math.abs(a), Math.abs(b)));
+  const upper = Math.floor(Math.max(Math.abs(a), Math.abs(b)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+};
+getRandomPositiveInteger(1, 8);
+
+const getRandomPositiveFloat = (a, b, digits = 1) => {
+  const lower = Math.min(Math.abs(a), Math.abs(b));
+  const upper = Math.max(Math.abs(a), Math.abs(b));
+  const result = Math.random() * (upper - lower) + lower;
+  return +result.toFixed(digits);
+};
+getRandomPositiveFloat(1.3, 4.9, 3);
+
+const getRandomArrayElement = (elements) => elements[getRandomPositiveInteger(0, elements.length - 1)];
+
+const makeRandomSelection = (count) => {
+  const newFeatures = [];
+  if (count >= FEATURES_ITEMS.length) {
+    return FEATURES_ITEMS;
   }
-  const degree = 10 ** maxDigits;
-  return ~~ ((Math.random() *(max - min) + min) * degree) / degree;
-}
-getRandomFloating(1, 4, 3);
+  for (let i = 0; i < count; i++) {
+    let newElement = FEATURES_ITEMS[Math.floor(Math.random() * FEATURES_ITEMS.length)];
+    while (newFeatures.includes(newElement)) {
+      newElement = FEATURES_ITEMS[Math.floor(Math.random() * FEATURES_ITEMS.length)];
+    }
+    newFeatures.push(newElement);
+  }
+  return newFeatures;
+};
+
+
+const makeTitles = () => ({
+  author: {
+    avatar: '',//Тут не понимаю как можно сделать
+  },
+  offer: {
+    title: offerTitle,
+    adress: ' ',// здесь тоже провал
+    price: getRandomArrayElement(PRICE_ROOM),
+    type: getRandomArrayElement(TYPE_OF_ROOMS),
+    rooms: getRandomArrayElement(ROOMS_COUNT),
+    guests: getRandomArrayElement(GUESTS_COUNT),
+    checkin: getRandomArrayElement(CHEK_IN_COUNTER),
+    checkout: getRandomArrayElement(CHECK_OUT_COUNTER),
+    features: makeRandomSelection(3), // здесь я тоже не уверен, нашел функцию в интернете!
+    description: descriptionRoom,
+    photos: getRandomArrayElement(ROOM_PHOTOS),
+  },
+  location: {
+    lat: getRandomPositiveFloat(locationLat[1], locationLat[2], locationLat[3]),
+    lng: getRandomPositiveFloat(locationLng[1], locationLng[2], locationLng[3]),
```

Comment:
Третий аргумент лучше вынести в отдельную константу общую, например как LOCATION_DIGITS или что-то вроде этого

Reply:
Иначе сложно понять, что означают числа 1 2 3 пока не взглянешь на содержимое переменной которые так же можно заменить как
```suggestion
    lng: getRandomPositiveFloat(MIN_LNG, MAX_LNG, LOCATION_DIGITS),
```

Желательно вообще стремиться к тому, чтобы код читался как текст